full_build_upgraded/full_build/self_loop_optimized.py

```
@@ -0,0 +1,255 @@
+        # Wait for expanded retrieval to complete if it was started
+        if expanded_retrieval_task:
+            try:
+                new_docs = await expanded_retrieval_task
+                if new_docs:
+                    ctx_docs = new_docs
+                    ctx_text = "\n\n".join(
+                        f"[{i+1}] {doc.text if hasattr(doc, 'text') else str(doc)}"
+                        for i, doc in enumerate(ctx_docs)
+                    )
+            except Exception:
+                pass  # Use existing context
```

Ruff (0.13.3)

`try`-`except`-`pass` detected, consider logging the exception

(S110)